databases/week1/dataBaseweek1.sql.txt

```diff
@@ -0,0 +1,59 @@
+           /*Homework week1 Tariq Javid*/
+/* Find out how many tasks are in the task table*/
+SELECT count(*) as totalTask from task;
+
+
+/* Find out how many tasks in the task table do not have a valid due date*/
+SELECT  count(*) FROM task
+ where due_date is null; 
+
+
+/*Find all the tasks that are marked as done*/
+SELECT * FROM task 
+WHERE status_id = 3;
+
+/*Find all the tasks that are not marked as done*/
+SELECT * FROM task 
+WHERE status_id !=3;
+
+
+/*Get all the tasks, sorted with the most recently created first*/
+SELECT *  from task
+order by   created DESC;
+
+/*Get the single most recently created task*/
+SELECT *  from task
+order by   created DESC
+limit 1;
+
+
+/*Get the title and due date of all tasks where the title or description contains database*/
+SELECT  title, due_date from task
+where title like '%database%';
+
+
+/*Get the title and status (as text) of all tasks*/
+SELECT T.title, S.name FROM task as T
+INNER JOIN status as S  ON T.status_id = S.id;
+
+
+/*Get the name of each status, along with a count of how many tasks have that status*/
+SELECT S.name ,count(*)as "No of Task"  FROM task as T
+left JOIN status as S  ON T.status_id = S.id
+Group by  S.id; 
+
+
+/*Get the names of all statuses, sorted by the status with most tasks first*/
+SELECT T.id,S.name,T.created from task as T
+ LEFT JOIN status as S  ON T.status_id = S.id
+ order by T.created DESC;
```

Comment:
This one is not quite there. You should do the query exactly like the previous above, except also order the result by the count